lib/authentication/secure_storage/json_credential_manager.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2015-2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+const path = require('path');
+const Logger = require('../../logger');
+const fs = require('fs');
+const os = require('os');
+const Util = require('../../util');
+
+function JsonCredentialManager() {
+  this.getTokenDir = function () {
+    let tokenDir = process.env.SF_TEMPORARY_CREDENTIAL_CACHE_DIR;
+    if (!Util.exists(tokenDir)) {
+      tokenDir = os.homedir();
+    }
+    if (!Util.exists(tokenDir)) {
+      Logger.getInstance().error(`Temporary credential cache directory is invalid, and the driver is unable to use the default location(home). 
+      Please assign the enviroment variable value SF_TEMPORARY_CREDENTIAL_CACHE_DIR to enable the default credential manager.`);
+      return null; 
+    }
+    const tokenCacheFile = path.join(tokenDir, 'temporary_credential.json');
+    Logger.getInstance().info('Cache directory: ', tokenCacheFile);
+    return tokenCacheFile;
+  };
+
+  this.tokenDir = this.getTokenDir();
+  
+  this.readJsonCredentialFile = function () {
+    const cred = fs.readFileSync(this.tokenDir, 'utf8');
+    try {
+      return JSON.parse(cred);
+    } catch (err) {
+      return null;
+    }
+  };
+  
+  this.write = function (key, token) {
+    if (!Util.checkParametersDefined(key) || !this.tokenDir) {
+      return null;
+    }
+    let jsonCredential = null;
+  
+    if (fs.existsSync(this.tokenDir)){
+      jsonCredential = this.readJsonCredentialFile() || {};
+    } else {
+      jsonCredential = {}; 
+    }
+    jsonCredential[key] = token;
+   
+    fs.writeFileSync(this.tokenDir, JSON.stringify(jsonCredential), 'utf8');
```

Comment:
This is the place where we also create a file in case it doesn't exist? If so, then what are the read/write access rights of this file? It should only be accessed by the account user.

Reply:
I added the mode which makes the account users only read and write.